api/tests/controllers/recipes.test.js

```diff
@@ -1,45 +1,229 @@
-const axios = require("axios");
-const puppeteer = require("puppeteer");
+const app = require("../../app");
+const Recipe = require("../../models/recipe");
+const User = require("../../models/user");
+const request = require("supertest");
+const JWT = require("jsonwebtoken");
+require("../mongodb_helper");
 
-const  RecipesController = require("../../controllers/recipes");
+const secret = process.env.JWT_SECRET;
+let recipe;
 
-jest.mock("axios");
-jest.mock("puppeteer");
+const createToken = (userId) => {
+  return JWT.sign(
+    {
+      user_id: userId,
+      // Backdate this token of 5 minutes
+      iat: Math.floor(Date.now() / 1000) - 5 * 60,
+      // Set the JWT token to expire in 10 minutes
+      exp: Math.floor(Date.now() / 1000) + 10 * 60,
+    },
+    secret
+  );
+};
 
-describe("RecipesController", () => {
-    it("should fetch recipe data successfully", async () => {
-    const mockAxiosResponse = {
-    data: "<html><body>Mock HTML content</body></html>",
-    };
-
-    axios.get.mockResolvedValueOnce(mockAxiosResponse);
-    puppeteer.launch.mockResolvedValueOnce({
-    newPage: jest.fn().mockResolvedValueOnce({
-        goto: jest.fn().mockResolvedValueOnce(),
-        $$eval: jest.fn().mockImplementation((selector, callback) => callback([])),
-    }),
-        close: jest.fn().mockResolvedValueOnce(),
+let token;
+let decodedToken;
+describe("/recipes", () => {
+  beforeAll(async () => {
+    // add user
+    const user = new User({
```

Comment:
There's a way to create a user to make a userId for creating token without having to actually write a new user into the database. Could be refactored later if have time.